### ISSUE TITLE
ReverseMap creates object for navigation property

### ISSUE THREAD

**Issue by danielgreen** [AutoMapper/AutoMapper#2151]
### Source/destination types

```csharp
public partial class TimesheetModel
{
        public  int ID { get; set; }
        public  DateTime? StartDate { get; set; }
        public  int? Contact { get; set; }
        public  ContactModel ContactNavigation { get; set; }
}

public class TimesheetViewModel
{
        public int? Contact { get; set; }
        public DateTime? StartDate { get; set; } = DateTime.Now;
}
```

### Mapping configuration

```csharp
CreateMap<TimesheetModel, TimesheetViewModel>().ReverseMap();
```

However, if I use the following mapping configuration, without ReverseMap, the issue does not occur:
```csharp
CreateMap<TimesheetModel, TimesheetViewModel>();
CreateMap<TimesheetViewModel, TimesheetModel>();
```

### Version: x.y.z
AutoMapper 6.1.0, netcoreapp 1.1, EF Core 1.1.2

### Expected behavior
In 6.0.2, when mapping from TimesheetViewModel to TimesheetModel (i.e. via the reverse map), the TimesheetModel object would be created with ID = 0, the Contact and StartDate values copied over, and the ContactNavigation property set to null.

### Actual behavior
In 6.1.0, the ContactNavigation property is set to an instance of ContactModel, presumably from a call to new ContactModel() within AutoMapper. No values are set for any property of ContactModel.

When the TimesheetModel is subsequently added to the EF Core db context, EF also sees the related Contact entity as new adds it to the change tracker. When attempt to save, it fails because of some non-nullable columns on the Contact table that we have not set values for on the Contact entity.

### Steps to reproduce

```csharp
// timesheetViewModel constructed by MVC model binding
var timesheetModel = _mapper.Map<TimesheetModel>(timesheetViewModel);
```

**Comment by lbargaoanu** [comment]
A repro would help. Make a [gist](https://gist.github.com/) that we can execute and see fail.

**Comment by danielgreen** [comment]
Could you point to an example of such a gist for AutoMapper please?

**Comment by lbargaoanu** [comment]
:) Smth like [this](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9).

**Comment by FilipFilipov** [comment]
I can confirm the same happens when you use MapFrom to fill a property from a related entity. Please give users the option to keep the old behavior.

**Comment by jbogard** [comment]
ReverseMap is now assuming you want to perform a reverse map, including un-flattening. However if this is a new issue, please open a new one.

**Comment by FilipFilipov** [comment]
Well it's assuming wrong. When I reverse map a create view model to a new database entity with a foreign key, why would I want it to have an invalid related entity that is already present in my database tacked on it?

**Comment by lbargaoanu** [comment]
@FilipFilipov A repro would help. Make a [gist](https://gist.github.com/lbargaoanu/9c7233441c3a3413cc2b9b9ebb5964a9) that we can execute and see fail.

**Comment by jbogard** [comment]
Then don't call ReverseMap. It's now a special kind of map designed for
reverse mapping into an existing object.

Which is what probably 99% of people really wanted in ReverseMap. You're
the (unfortunately) 1% that we thought we might break but weren't entirely
sure.

**Comment by FilipFilipov** [comment]
I'm not following. Even if you map into an existing object, most people wouldn't want to update its entire object graph with it, which is what entity framework tries to do when save the changes. Can't you just give ReverseMap a boolean parameter for this or something?

**Comment by jbogard** [comment]
No, that's not the case. Most people have the destination type as a subset of the source type, so I'm already only mapping a subset of properties back in. And if you don't want to map something, there's ForPath.Ignore

**Comment by lbargaoanu** [comment]
A workaround is to use ResolveUsing instead, but that doesn't work with ProjectTo.

**Comment by jbogard** [comment]
Is there a better repro? I still dont see it.

**Comment by lbargaoanu** [comment]
Why would they bother? :) I made [one](https://gist.github.com/lbargaoanu/908eb76ad493ee176a6b7a45d903ef8d).

**Comment by protonyx** [comment]
@jbogard We were using ReverseMap() as a shorthand so we didn't have to create another map explicitly, but now the behavior has changed to include un-flattening logic. Though I agree that most cases would want to have this logic included, it does represent a breaking change in the behavior of ReverseMap. It might be worth adding a warning in the release notes.

**Comment by lbargaoanu** [comment]
It should be easy to create an extension method that has the old behavior. The release notes link to the [docs](https://github.com/AutoMapper/AutoMapper/wiki/Reverse-Mapping-and-Unflattening).

**Comment by jgidlow** [comment]
We encountered a similar problem with ReverseMap in 6.1.0 that made us have to revert back to 6.0.2. 

When calling IMapper.Map(source, destination) for a source model without a collection property the collection property on the destination model got set to null after the call to Map. I'm guessing just because the property didn't exist on the source model?

In 6.0.2 this does not happen and just copies  over the properties that exists on the source model and leaves the other properties alone on the destination model.

When I tried to write a repro for this with non EF models I didn't manage to reproduce it though even to the mappings were the same.

If this behaviour won't get added back somehow we would have to remove all our ReverseMap mappings which would be a shame because I really liked how easy and intuitive that setting was.
